lib/model/connection.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * connection.js
+ *
+ * Definition for the Connection class. A Connection is the linkage between a
+ * Node's Output and another (potentially different) node's Input.
+ * 
+ * Alternatively, in the situation where a node is contained within another node,
+ * the containee can have a connection from one of it's Outputs to one of the
+ * container's Outputs. The same applies for Inputs.
+ */
+
+/**
+ * Constructor function.
+ *
+ * @param id - The Connection's unique identifier.
+ * @param from - The Output object that is the connection's source.
+ * @param to - The Input object that is the connection's target.
+ * @param metadata - A standard JavaScript object containing key-value metadata.
+ */
+var Connection = function(id, from, to, metadata) {
+	this.id = id || 0;
```

Comment:
Everything being set in all these classes. At the very least make sure they are the right data type and defined.